### PR TITLE
feat: add conformance tests 

### PR DESCRIPTION
**Pull request by dunglas**

### CODE REVIEW

File: conformance-tests/mercure.spec.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+function randomString() {
+  return randomBytes(20).toString('hex');
+}
+
+test.beforeEach(async ({ page }) => await page.goto('/'));
+
+test.describe('Publish update', () => {
+  const randomStrings: string[] = [];
+  for (let i = 0; i < 6; i++)
+    randomStrings[i] = randomString();
```
Comment (soyuka) [inline conformance-tests/mercure.spec.ts:13]:
```suggestion
  const randomStrings: string[] = Array(5).fill('').map(randomString)
```

File: conformance-tests/mercure.spec.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
+import { test, expect } from '@playwright/test';
+import { randomBytes } from "crypto";
+
+function randomString() {
```
Comment (soyuka) [inline conformance-tests/mercure.spec.ts:4]:
```suggestion
function randomString(): string {
```

File: conformance-tests/playwright.config.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  ],
+
+  /* Folder for test artifacts such as screenshots, videos, traces, etc. */
+  // outputDir: 'test-results/',
+
+  /* Run your local dev server before starting the tests */
+  // webServer: {
+  //   command: 'npm run start',
+  //   port: 3000,
+  // },
```
Comment (soyuka) [inline conformance-tests/playwright.config.ts:103]:
```suggestion
  ]
```

File: conformance-tests/mercure.spec.ts (hunk 1 above)
Comment (dunglas) [reply conformance-tests/mercure.spec.ts:13]: Isn't it slower and less readable?

File: conformance-tests/mercure.spec.ts (hunk 2 above)
Comment (dunglas) [reply conformance-tests/mercure.spec.ts:4]: This looks inferred automatically.

File: conformance-tests/mercure.spec.ts (hunk 1 above)
Comment (Fabious) [reply conformance-tests/mercure.spec.ts:13]:
```suggestion
  const randomStrings: string[] = Array.from({length: 6}, randomString);
```
Elegant way IMO

File: conformance-tests/mercure.spec.ts [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+      const { received, contentType, status, body } = await page.evaluate(async (data) => {
+        const receivedResult = Symbol('received');
+        const notReceivedResult = Symbol('not received');  
+
+        let resolveReady: Function;
+        const ready = new Promise((resolve) => {
+          resolveReady = function () {
+            resolve(true);
+          };
+        });
```
Comment (Fabious) [inline conformance-tests/mercure.spec.ts:46]:
```suggestion
        let resolveReady: () => void;
        const ready = new Promise((resolve) => {
          resolveReady = () => resolve(true)
        });
```
Minor type and syntax update, can be applied to `resolveReceived` below too if you like it

File: conformance-tests/mercure.spec.ts [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+        const url = new window.URL('/.well-known/mercure', window.origin);
+        data.topicSelectors.forEach(topicSelector => url.searchParams.append('topic', topicSelector));
+
+        const event = new window.URLSearchParams();
+        data.updateTopics.forEach(updateTopic => event.append('topic', updateTopic));
+        event.set('id', data.updateID);
+        event.set('data', `data for
+
+        ${data.name}`);
```
[the remark this answers is not in the document]
Comment (dunglas) [reply conformance-tests/mercure.spec.ts:63]: Having a blank line in the data is part of the test.

**Pull request merged**